sound/soc/intel/common/soc-acpi-intel-ptl-match.c

```diff
@@ -134,6 +152,12 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_ptl_sdw_machines[] = {
 		.drv_name = "sof_sdw",
 		.sof_tplg_filename = "sof-ptl-rt722.tplg",
 	},
+	{
+		.link_mask = BIT(1),
+		.links = ptl_rt722_l1,
+		.drv_name = "sof_sdw",
+		.sof_tplg_filename = "sof-ptl-rt722-4ch.tplg",
```

Comment:
From where this topology is coming and what is the `-4ch` postfix means? Is it PCH-DMIC? Should not that be dynamically added to the base topology name based on the number of DMICs detected?

Reply:
Yes, the "-4ch" should be added automatically. See https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/intel/hda.c#L1321. @RDharageswari Can you test if "-4ch" is added automatically on your device?

Reply:
as discussed on chat, -4ch .DMIC sting can be appended through module params. Will post the topology changes for 4ch dmic soon.

Reply:
The topology PR is https://github.com/thesofproject/sof/pull/9493

Reply:
the topology is wrong @bardliao, it relies on 2ch.